daft/table/table_io.py

```diff
@@ -108,40 +89,22 @@ def read_json(
     Returns:
         MicroPartition: Parsed MicroPartition from JSON
     """
-    io_config = None
-    if storage_config is not None:
-        config = storage_config.config
-        if isinstance(config, NativeStorageConfig):
-            assert isinstance(file, (str, pathlib.Path)), "Native downloader only works on string inputs to read_json"
-            json_convert_options = JsonConvertOptions(
-                limit=read_options.num_rows,
-                include_columns=read_options.column_names,
-                schema=schema._schema if schema is not None else None,
-            )
-            json_parse_options = JsonParseOptions()
-            tbl = MicroPartition.read_json(
-                str(file),
-                convert_options=json_convert_options,
-                parse_options=json_parse_options,
-                read_options=json_read_options,
-                io_config=config.io_config,
-            )
-            return _cast_table_to_schema(tbl, read_options=read_options, schema=schema)
-        else:
-            assert isinstance(config, PythonStorageConfig)
-            io_config = config.io_config
-
-    with _open_stream(file, io_config) as f:
-        table = pajson.read_json(f)
-
-    if read_options.column_names is not None:
-        table = table.select(read_options.column_names)
-
-    # TODO(jay): Can't limit number of rows with current PyArrow filesystem so we have to shave it off after the read
-    if read_options.num_rows is not None:
-        table = table[: read_options.num_rows]
-
-    return _cast_table_to_schema(MicroPartition.from_arrow(table), read_options=read_options, schema=schema)
+    config = storage_config if storage_config is not None else StorageConfig(True, IOConfig())
+    assert isinstance(file, (str, pathlib.Path)), "Native downloader only works on string inputs to read_json"
+    json_convert_options = JsonConvertOptions(
+        limit=read_options.num_rows,
+        include_columns=read_options.column_names,
+        schema=schema._schema if schema is not None else None,
+    )
+    json_parse_options = JsonParseOptions()
+    tbl = MicroPartition.read_json(
+        str(file),
+        convert_options=json_convert_options,
+        parse_options=json_parse_options,
+        read_options=json_read_options,
+        io_config=config.io_config,
+    )
+    return _cast_table_to_schema(tbl, read_options=read_options, schema=schema)
 
 
 def read_parquet(
```

Comment:
What do we need `table_io.read_parquet/csv/json` for? Is it possible to get rid of them even?

Reply:
You're right, I believe we eliminate the Python code path entirely. I think I'll leave that for a future PR and add a TODO here.

Reply:
In fact we could probably remove table_io.py entirely if we port our non-swordfish code to use `daft.io.writer` to do writing.